consensus-types/forks/eip4844/tx.go

```diff
@@ -14,13 +15,14 @@ func TxPeekBlobVersionedHashes(tx []byte) ([][32]byte, error) {
 	if tx[0] != types.BlobTxType {
 		return nil, errInvalidBlobTxType
 	}
-	// TODO(EIP-4844): bounds checking
-	offset := 1 + binary.LittleEndian.Uint32(tx[1:5])
-	hashesOffset := binary.LittleEndian.Uint32(tx[offset+156 : offset+160])
-	hashes := make([][32]byte, (uint32(len(tx))-hashesOffset)/32)
-	for i := hashesOffset; i < uint32(len(tx)); i += 32 {
+	sbt := types.SignedBlobTx{}
+	if err := sbt.Deserialize(codec.NewDecodingReader(bytes.NewReader(tx), uint64(len(tx)))); err != nil {
```

Comment:
It should be `tx[1]` and `len(tx)-1` here as the first-byte transaction type is not part of the SSZ.